packages/workspace/src/browser/workspace-service.ts

```diff
@@ -40,10 +40,10 @@ import { IJSONSchema } from '@theia/core/lib/common/json-schema';
 @injectable()
 export class WorkspaceService implements FrontendApplicationContribution {
 
-    private _workspace: FileStat | undefined;
+    protected _workspace: FileStat | undefined;
```

Comment:
I believe `_workspace` can remain `private` as it is accessible through:

```ts
get workspace(): FileStat | undefined {
    return this._workspace;
}
```

Same is true for other properties. @kenneth-marut-work when submitting https://github.com/eclipse-theia/theia/pull/9597 I believe you were not referring to `private` members accessible through getters, correct?

Reply:
So the issue that we frequently run into is that when overriding a class that includes private members, referencing that member from the override gives us errors, i.e:
![image](https://user-images.githubusercontent.com/62660714/122444260-fa204f00-cf65-11eb-890b-06961695e1af.png)

In the above case, it's true that we can reference the getter `workspace` variable since we are only reading the value, but in a case where we might want to override some setting behavior like this:

![image](https://user-images.githubusercontent.com/62660714/122443871-9bf36c00-cf65-11eb-9433-235e32cda351.png)

Neither the private `_workspace` or the public `workspace` getter will suffice. We either need an additional `setter`, or to just have the member be `protected` in the first place. Likely having the member protected is best since we might only want the value to be set internally by the class and not be public.

Reply:
@kenneth-marut-work thank you for clarifying, and I agree that `protected` would be better in such a case. It might be something we should think of enforcing in the framework (coding guidelines) to promote improved extensibility.

Reply:
No problem, and I agree :)